hive_flutter/lib/src/adapters/time_adapter.dart

```diff
@@ -1,16 +1,18 @@
-import 'package:flutter/material.dart';
-import 'package:hive/hive.dart';
+part of '../../adapters.dart';
 
-class TimeAdapter extends TypeAdapter<TimeOfDay> {
+class TimeOfDayAdapter extends TypeAdapter<TimeOfDay> {
   @override
   TimeOfDay read(BinaryReader reader) {
-    final time = reader.read() as TimeOfDay;
-    return time;
+    return TimeOfDay(
+      hour: reader.readInt(),
+      minute: reader.readInt(),
+    );
```

Comment:
I think we could merge minutes and hours together like this. What do you think about that?

```suggestion
    final totalMinutes = reader.readInt();
    return TimeOfDay(hour: totalMinutes /~ 60, minute: totalMinutes % 60);
```